control-plane-operator/controllers/hostedcontrolplane/autoscaler/reconcile.go

```diff
@@ -95,7 +95,8 @@ func ReconcileAutoscalerDeployment(deployment *appsv1.Deployment, hcp *hyperv1.H
 						Name: "target-kubeconfig",
 						VolumeSource: corev1.VolumeSource{
 							Secret: &corev1.SecretVolumeSource{
-								SecretName: kubeConfigSecret.Name,
+								SecretName:  kubeConfigSecret.Name,
+								DefaultMode: k8sutilspointer.Int32Ptr(416),
```

Comment:
Can we use octal here?  416 means nothing to me, 0640 is better.  Obviously for all of them.

Reply:
@imain our aim is to avoid global read so not sure if that will work  cc@hasueki @relyt0925

Reply:
@kimetro

`0640` is just the octal representation of same number, decimal `416`. Looks like the API accepts either values, so more of a preference more than anything.

>// defaultMode is Optional: mode bits used to set permissions on created files by default.
	// Must be an octal value between 0000 and 0777 or a decimal value between 0 and 511.
	// YAML accepts both octal and decimal values, JSON requires decimal values
	// for mode bits. Defaults to 0644.
	// Directories within the path are not affected by this setting.
	// This might be in conflict with other options that affect the file
	// mode, like fsGroup, and the result can be other mode bits set.
	// +optional
	DefaultMode *[int32](https://pkg.go.dev/builtin#int32) `json:"defaultMode,omitempty" protobuf:"bytes,3,opt,name=defaultMode"`

Reply:
yes, and the chmod command takes octal and is commonly used.  Octal is used because each single digit used aligns with the given permission for user/group/all etc.  It's a defacto standard really and most people who use unix systems can tell exactly what an octal permission is and how it applies.  Not the case for decimal.  If octal works it would make the code much more readable.

Thanks! :)